### PR TITLE
Flash payload

### PR DESCRIPTION
**Pull request by UKNOWH**
## About the PR 
Added a flash payload which can be inserted into a modular grenade , that functions as a flashbang
**Screenshots**

**Changelog**

:cl:
- add: Added a Flash payload that allows players to make modular flashbangs

### CODE REVIEW

**Comment by Fishfish458** [comment]
There really should be a chemical reaction that causes flash, though can probably live in tandem with an actual charge that gets printed.

File: Resources/Prototypes/Recipes/Lathes/devices.yml [hunk 1]
```diff
@@ -19,3 +19,16 @@
...
     Steel: 200
     Plastic: 300
+
+- type: latheRecipe
+  id: FlashPayload
+  icon:
+    sprite: Objects/Devices/payload.rsi
+    state: payload-flash-armed
+  result: FlashPayload
+  completetime: 200
```
Comment (ElectroJr) [inline Resources/Prototypes/Recipes/Lathes/devices.yml:29]: lathes were recently updated so that the completetime is in seconds, so this should probably just be `completetime: 1`

File: Resources/Prototypes/Entities/Objects/Devices/payload.yml [hunk 2]
```diff
@@ -76,3 +76,19 @@
         enum.ChemicalPayloadFilledSlots.Left: payload-chemical-left
         enum.ChemicalPayloadFilledSlots.Right: payload-chemical-right
         enum.ChemicalPayloadFilledSlots.Both: payload-chemical-armed
+
+- type: entity
+  name: flash payload
+  parent: BasePayload
+  id: FlashPayload
+  description: a flash payload , it only have one use
```
Comment (ElectroJr) [inline Resources/Prototypes/Entities/Objects/Devices/payload.yml:84]: Grammar wise this should be "has" instead of "have", but maybe just reword it to something like "A single-use flash payload.", which is a bit more concise.

**Review by ElectroJr — CHANGES_REQUESTED** [review]
Other than those minor comments, seems all fine.

Though IMO the modular flash grenade should also delete the casing & trigger when "exploding", not just the payload. But implementing that requires adding some c# code & probably adding a new component. Unless you're happy to try add that I can do that in a separate PR.

**Comment by UKNOWH** [comment]
noted , but deleting the casing kind of defeted the recyclability of the modular grenade. And even the strongest chem explosion could not destroy the casing , I don't see why the flash would

File: Resources/Prototypes/Recipes/Lathes/devices.yml (hunk 1 above)
Comment (UKNOWH) [reply Resources/Prototypes/Recipes/Lathes/devices.yml:29]: noted

File: Resources/Prototypes/Entities/Objects/Devices/payload.yml (hunk 2 above)
Comment (UKNOWH) [reply Resources/Prototypes/Entities/Objects/Devices/payload.yml:84]: I will change that

**Comment by ElectroJr** [comment]
The chem-explosions definitely used to, but that was before the explosion refactor which probably made them too weak to break them. Explosive reactions definitely should break grenades. But I'm, open to the idea that flash stuff could be re-usable. Though the grenade visuals definitely have to be updated eventually so that people can differentiate a "spent" grenade from one that is till active. I guess its fine to just leave it as is for now

File: Resources/Prototypes/Recipes/Lathes/devices.yml [hunk 3]
```diff
@@ -19,3 +19,16 @@
...
+  icon:
+    sprite: Objects/Devices/payload.rsi
+    state: payload-flash-armed
+  result: FlashPayload
+  completetime: 1
+  materials:
+    Steel: 25
+    Plastic: 50
+    Glass: 25
+    #one fourth of what making a flash would cost
```
Comment (mirrorcult) [inline Resources/Prototypes/Recipes/Lathes/devices.yml:34]: this should be quite a bit more expensive imo, shouldnt be this easy to essentially mass manufacture flashbangs

**Review by mirrorcult — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Recipes/Lathes/devices.yml (hunk 3 above)
Comment (UKNOWH) [reply Resources/Prototypes/Recipes/Lathes/devices.yml:34]: sure , increase it by 2 fold ? I maded it a fourth of a chem payload.
